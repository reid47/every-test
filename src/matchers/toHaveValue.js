import { formatValue } from './common';

export default function toHaveValue(received, expectedValue) {
  const actualValue = received.value;
  const pass = actualValue === expectedValue;
  const message = pass
    ? `Expected element not to have value ${formatValue(expectedValue)}.`
    : `Expected element to have value ${formatValue(
        expectedValue
      )}, but actual value was ${formatValue(actualValue)}.`;

  return { pass, message };
}
