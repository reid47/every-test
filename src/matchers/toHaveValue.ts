import { formatValue } from './common';

export function toHaveValue(received, expectedValue) {
  const actualValue = received.value;
  const pass = actualValue === expectedValue;

  const message = pass
    ? `Expected element not to have value ${formatValue(expectedValue)}.`
    : `Expected element to have value ${formatValue(
        expectedValue
      )}, but actual value was ${formatValue(actualValue)}.`;

  return { pass, message };
}
