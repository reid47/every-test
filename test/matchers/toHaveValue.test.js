import toHaveValue from '../../src/matchers/toHaveValue';

describe('toHaveValue', () => {
  test('passing', () => {
    const el = document.createElement('input');
    el.value = 'hello';

    const result = toHaveValue(el, 'hello');
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const el = document.createElement('input');
    el.value = 'hello';

    const result = toHaveValue(el, 'something else');
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
