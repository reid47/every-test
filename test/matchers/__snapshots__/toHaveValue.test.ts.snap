// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toHaveValue failing 1`] = `"Expected element to have value \\"something else\\", but actual value was \\"hello\\"."`;

exports[`toHaveValue passing 1`] = `"Expected element not to have value \\"hello\\"."`;
